contrib/babelfishpg_common/src/varchar.c

```diff
@@ -1188,6 +1191,84 @@ fixeddecimal2bpchar(PG_FUNCTION_ARGS)
 #undef FIXEDDECIMAL_2_VARCHAR_MULTIPLIER
 #undef FIXEDDECIMAL_2_VARCHAR_SCALE
 
+Datum
+float82varchar(PG_FUNCTION_ARGS)
+{
+	float8 num = PG_GETARG_FLOAT8(0);
+	int32 typmod = PG_GETARG_INT32(1);
+	int maxlen = (typmod == -1) ? 0 : (typmod - VARHDRSZ);
+	Datum res;
+	char	   *ascii = (char *) palloc(32);
+	
+	/* round to 6 decimal digits */
+	if (!(isinf(num)|| isnan(num))) {
+		num = round(num * 1000000.0) / 1000000.0;
+	}
+
+	double_to_shortest_decimal_buf(num, ascii);
+
+	/* Check if the number fits within the specified length */
+	if (maxlen > 0) {
+		size_t str_len = strlen(ascii);
+		if (str_len > maxlen) {
+			ereport(ERROR,
+					(errcode(ERRCODE_STRING_DATA_RIGHT_TRUNCATION),
+					 errmsg("There is insufficient result space to convert a float value to varchar/nvarchar.")));
+		}
+	}
+	
+	res = DirectFunctionCall3(varcharin,
+							   CStringGetDatum(ascii),
+							   ObjectIdGetDatum(0),
+							   Int32GetDatum(-1));
```

Comment:
lets pass the typmod we received in this function

Reply:
Fixed in latest revision